components/permission-controller/controllers/namespace_controller_test.go

```diff
@@ -0,0 +1,221 @@
+				close(stopMgr)
+				mgrStopped.Wait()
+			})
+
+			It("not add static user to subjects if UseStaticConnector is set to false", func() {
+
+				testRolebindingName, testNamespaceName := controllers.RolebindingName, "dev-static"
+				expectedRequest := &reconcile.Request{NamespacedName: types.NamespacedName{Name: testNamespaceName}}
+
+				s := scheme.Scheme
+				err := rbac.AddToScheme(s)
+				Expect(err).NotTo(HaveOccurred())
+
+				err = corev1.AddToScheme(scheme.Scheme)
+				Expect(err).NotTo(HaveOccurred())
+
+				// Setup Manager and Controller.  Wrap the Controller Reconcile function so it writes each request to a
+				// channel when it is finished.
+				mgr, err := manager.New(cfg, manager.Options{Scheme: s})
+				Expect(err).NotTo(HaveOccurred())
+				c := mgr.GetClient()
+
+				reconcileFn, requests := SetupTestReconcile(getAPIReconciler(mgr, false))
+				Expect(add(mgr, reconcileFn)).To(Succeed())
+
+				// Start the manager and the controller
+				stopMgr, mgrStopped := StartTestManager(mgr)
+
+				testNamespace := &corev1.Namespace{
+					ObjectMeta: v1.ObjectMeta{
+						Name: testNamespaceName,
+					},
+				}
+				err = c.Create(context.TODO(), testNamespace)
+				Expect(err).NotTo(HaveOccurred())
+				Eventually(requests, timeout).Should(Receive(Equal(*expectedRequest)))
+
+				// Verify if the rolebinding is created
+				expectedSubjects := []rbac.Subject{
+					{
+						Kind:     "Group",
+						Name:     adminGroup,
+						APIGroup: rbac.GroupName,
+					},
+				}
+				var retrieved rbac.RoleBinding
+				ok := client.ObjectKey{Name: testRolebindingName, Namespace: testNamespaceName}
+				err = c.Get(context.TODO(), ok, &retrieved)
+				fmt.Println(retrieved.Subjects)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(retrieved.RoleRef.Name).To(Equal(controllers.RoleRefName))
+				Expect(retrieved.RoleRef.Kind).To(Equal(controllers.RoleRefKind))
+				Expect(retrieved.Subjects).To(Equal(expectedSubjects))
+
+				c.Delete(context.TODO(), testNamespace)
+				close(stopMgr)
+				mgrStopped.Wait()
+			})
+		})
+
+		It("not create a rolebinding if the namespace is a system one", func() {
+
+			testRolebindingName, testNamespaceName := controllers.RolebindingName, systemNamespace
+			expectedRequest := &reconcile.Request{NamespacedName: types.NamespacedName{Name: testNamespaceName}}
+
+			s := scheme.Scheme
+			err := rbac.AddToScheme(s)
+			Expect(err).NotTo(HaveOccurred())
+
+			err = corev1.AddToScheme(s)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Setup Manager and Controller.  Wrap the Controller Reconcile function so it writes each request to a
+			// channel when it is finished.
+			mgr, err := manager.New(cfg, manager.Options{Scheme: s})
+			Expect(err).NotTo(HaveOccurred())
+			c := mgr.GetClient()
+
+			reconcileFn, requests := SetupTestReconcile(getAPIReconciler(mgr, true))
+			Expect(add(mgr, reconcileFn)).To(Succeed())
+
+			// Start the manager and the controller
+			stopMgr, mgrStopped := StartTestManager(mgr)
```

Comment:
This block is very similar in all three cases. It can be extracted to some common funtion.